extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java

```diff
@@ -164,6 +166,18 @@ public void build(RecorderContext recorder, HibernateOrmTemplate template,
         JpaJandexScavenger scavenger = new JpaJandexScavenger(reflectiveClass, descriptors, compositeIndex, nonJpaModelClasses);
         final JpaEntitiesBuildItem domainObjects = scavenger.discoverModelAndRegisterForReflection();
 
+        // register the strategies of @GenericGenerator for reflection
+        Set<String> strategies = new HashSet<>();
+        for (AnnotationInstance annotation : index.getIndex().getAnnotations(GENERIC_GENERATOR)) {
+            String strategy = annotation.value("strategy").asString();
+            if (strategy.contains(".")) {
```

Comment:
wondering: why checking for the dot being contained in the strategy string?

Reply:
If I am not mistakem, strategy could be a simple string, but maybe I am mixing this with the strategy from `GenericValue`?

Reply:
you're right it could be a special keyword, but also it could be a "fully qualified name" with no package. (That's far fetched but some people do it!)

We also need to deal with the opposite problem: custom identifier generators can be registered with any name, I'm not aware of a rule that prohibits these to have a "." in the name.

sorry I know that's all bad news but that's how it is :)

It might help to peek into `org.hibernate.id.factory.internal.DefaultIdentifierGeneratorFactory#getIdentifierGeneratorClass`. 

you could also - instead of registering these for reflection - instantiate the various identifierGenerators , store them in an immutable map, and insert a custom `IdentifierGeneratorFactory` implementation in the ORM bootstrap registry.

Reply:
Not a problem at all, I'll revisit later on tonight or over the weekend if you don't mind :)

Reply:
sure, thanks a lot!

Reply:
WRT the second approach IIUC I would have to change https://github.com/quarkusio/quarkus/blob/master/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/boot/registry/PreconfiguredServiceRegistryBuilder.java#L203 with an Initiator that would return the custom `IdentifierGeneratorFactory`, correct? I'm asking because I haven't looked at the Hibrnate part of Quarkus before ( and this PR is a great opportunity to do so :) ) and I don't want to stray down a completely incorrect path :P.